.dotnet/src/Generated/Models/ChatCompletionOptions.Serialization.cs

```diff
@@ -90,15 +90,17 @@ protected virtual void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWrit
                 }
                 writer.WriteEndArray();
             }
-            if (_additionalBinaryDataProperties?.ContainsKey("messages") != true)
+            // CUSTOM: Add a null check to allow Messages to behave like an optional
+            if (Messages is not null && _additionalBinaryDataProperties?.ContainsKey("messages") != true)
```

Comment:
Looking at how other optional collection properties work (e.g, `StopSequences`) for consistency, and if I'm not mistaken, it appears that the check here should be:

```csharp
Optional.IsCollectionDefined(StopSequences)
```

instead of

```csharp
Messages is not null
```

And then the change in `ChatCompletionOptions.Serialization.cs` can be reverted.

Should we also initialize the `Messages` property to an empty `ChangeTrackingList` for consistency?